stan/math/prim/prob/gaussian_dlm_obs_lpdf.hpp

```diff
@@ -112,15 +112,8 @@ gaussian_dlm_obs_lpdf(
     Eigen::Matrix<T_lp, Eigen::Dynamic, 1> m(n);
     Eigen::Matrix<T_lp, Eigen::Dynamic, Eigen::Dynamic> C(n, n);
 
-    // TODO(anyone): how to recast matrices
```

Comment:
Is it guaranteed at this point `m0` doesn't have scalar `T_lp`?  If it does, the copy's unnecessary.

Reply:
m0 can have scalar type T_lp. And m is modified later so we do need a copy anyway. If I'm understanding correctly.

Reply:
OK.